engine/apps/grafana_plugin/views/status.py

```diff
@@ -38,5 +38,6 @@ def get(self, _request: Request) -> Response:
                 "is_user_anonymous": self.grafana_context["IsAnonymous"],
                 "license": settings.LICENSE,
                 "version": settings.VERSION,
+                "currently_undergoing_maintenance_message": settings.CURRENTLY_UNDERGOING_MAINTENANCE_MESSAGE,
```

Comment:
Wouldn't this result in a 500 HTTP error in case DB is unreachable? There are some DB queries running before returning the response, maybe it's better to check for this setting before accessing the DB (including DB access in authentication classes)?

Reply:
This has me thinking about whether maintenance mode should be more explicit as in a different settings file or possibly introducing middleware instead of handling different API calls case by case.